appdatabase/migrations/sql/1691173699_add_collectibles_and_collections_data_cache.up.sql

```diff
@@ -0,0 +1,51 @@
+CREATE TABLE IF NOT EXISTS collectible_data_cache (
+    chain_id UNSIGNED BIGINT NOT NULL,
+    contract_address VARCHAR NOT NULL,
+    token_id BLOB NOT NULL,
+    provider VARCHAR NOT NULL,
+    name VARCHAR NOT NULL,
+    description VARCHAR NOT NULL,
+    permalink VARCHAR NOT NULL,
+    image_url VARCHAR NOT NULL,
+    animation_url VARCHAR NOT NULL,
+    animation_media_type VARCHAR NOT NULL,
+    background_color VARCHAR NOT NULL,
+    token_uri VARCHAR NOT NULL
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS collectible_data_identify_entry ON collectible_data_cache (chain_id, contract_address, token_id);
+
+CREATE TABLE IF NOT EXISTS collectible_traits_cache (
+    chain_id UNSIGNED BIGINT NOT NULL,
+    contract_address VARCHAR NOT NULL,
+    token_id BLOB NOT NULL,
+    trait_type VARCHAR NOT NULL,
+    trait_value VARCHAR NOT NULL,
+    display_type VARCHAR NOT NULL,
+    max_value VARCHAR NOT NULL,
+    FOREIGN KEY(chain_id, contract_address, token_id) REFERENCES collectible_data_cache(chain_id, contract_address, token_id) 
+      ON UPDATE CASCADE 
+      ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS collection_data_cache (
```

Comment:
How about using a ROWID as a collection ID and add an optional `collectionID` in `collectible_data_cache` for a more explicit collection association?

Reply:
hmm I left them separate because some providers have separate API endpoints for collectible and collection data, so I might be able to have one without having the other.

Reply:
oh that's why you said `optional`. Hmm is there a way I can link the "real" IDs somehow instead (that is, to say columns `chain_id` and `contract_address` from both collectibles and collections tables contain the same data)? If not I'll have to manually manage the new ID column every time I add a collectible/collection.

Reply:
Since collectible/collection metadata updates are independent, I'm gonna leave it as-is for now and merge this. I'll consider establishing a more elaborate relationship between the two later, if needed.